subprojects/packagefiles/libccp4c/meson.build

```diff
@@ -0,0 +1,104 @@
+# Meson file for libccp4c (CCP4 core libraries, C part only)
+project('libccp4c', ['c'],
+        version: '6.5.1',
+        meson_version: '>=0.49.0',
+        license: 'LGPL3')
+
+
+cc = meson.get_compiler('c')
+mdep = cc.find_library('m', required: true)
+
+
+# Note that the source code assumes that datadir='share'.
```

Comment:
You can assert this I suppose... how does it handle different prefixes?

Reply:
It literally does the following:
```
      env_file = ccp4_utils_joinfilenames(ccp4_utils_joinfilenames(
                  ccp4_utils_joinfilenames(pkgroot,"share"),"ccp4"),"environ.def");
```
Where `pkgroot` is set like this:
```
#ifdef PACKAGE_ROOT
  const char *pkgroot=PACKAGE_ROOT;
#else
  const char *pkgroot=get_parent_directory(argv[0]);
#endif
```
And `PACKAGE_ROOT` is set by meson.build.  However, this can be bypassed by setting an environment variable (`CINCL`).  This environment variable would be set when using the "full" CCP4 installation, but I wouldn't expect it to be the case when using the libraries via the wrap.

However, the environ.def (also default.def) file isn't part of these libraries, but rather part of the full CCP4 installation which is under a proprietary license.  It's just a list of default values for various program arguments, so it's a bit mean to not include it in the LGPL part, but that's the way it is.  Therefore, the question is kind of theoretical because the installed file will almost certainly never be seen when the wrap is used.  Adding an assertion seems like it would prevent the top-level project from customising the paths, without really adding any security because the project is almost certainly not using these files.  If it were, it would be part of the full CCP4 project and the environment variable would be set.

There is a related problem for some of the data files that are installed by the package, e.g. `syminfo.lib`.  The code will only be able to find them if an environment variable is set to the right location, even though the real location could easily be set by the build system (there is no particular reason to want to override the file).  This is something I _really_ want to fix via an upstream PR, but cannot yet do.

It's awful, I agree.  I don't see any way to fix it without changes upstream.

Reply:
Ah, okay -- thanks for explaining.